File: TFC-AppEventos/TFC.AppEventos.Application.DTO/Responses/LoginResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TFC.AppEventos.Transversal.Utils;

namespace TFC.AppEventos.Application.DTO.Responses
{
    public class LoginResponse : BaseResponse
    {
        public AuthDto AuthDto { get; set; }
        public UserDto User { get; set; }
        public string Token { get; set; }
        public DateTime TokenExpiration { get; set; }
    }
}

File: Flutter/app_eventos/lib/core/models/response/login_response.dart
import '../dto/auth_dto.dart';
import '../dto/user_dto.dart';

class LoginResponse {
  final AuthDto authDto;
  final UserDto user;
  final String token;
  final DateTime tokenExpiration;
  final bool isSuccess;
  final String? message;
  final int? responseCode;

  LoginResponse({
    required this.authDto,
    required this.user,
    required this.token,
    required this.tokenExpiration,
    required this.isSuccess,
    this.message,
    this.responseCode,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) => LoginResponse(
    authDto: AuthDto.fromJson(json['authDto']),
    user: UserDto.fromJson(json['user']),
    token: json['token'],
    tokenExpiration: DateTime.parse(json['tokenExpiration']),
    isSuccess: json['isSuccess'],
    message: json['message'],
    responseCode: json['responseCode'],
  );
  Map<String, dynamic> toJson() => {
    'authDto': authDto.toJson(),
    'user': user.toJson(),
    'token': token,
    'tokenExpiration': tokenExpiration.toIso8601String(),
    'isSuccess': isSuccess,
    'message': message,
    'responseCode': responseCode,
  };
}
